### PR TITLE
ATS-829  Release T-Engines 2.3.6

### PR DESCRIPTION
**Pull request by alandavis**
Linked to: REPO-5219 Allow AGS AMP to specify metadata extract mapping

Added an extractMapping transform option to all metadata extractors to override the default one.

### CODE REVIEW

**Review by cmshields — APPROVED** [review]
LGTM!

**Review by montgolfiere — APPROVED** [review]
I've had a quick look and look ok. 

Any updates to markdown design docs, re: metadata extraction ?

**Comment by montgolfiere** [comment]
Just as I ticked, I saw your update to Spring Boot 2.3.5 (see also ACS-492). I think 2.3.6 was also recently released (see jira comments).

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.3.1.RELEASE</version>
+        <version>2.3.5.RELEASE</version>
```
Comment (montgolfiere) [inline pom.xml:8]: JFYI: 2.3.6 now also available

**Review by eknizat — APPROVED** [review]
LGTM

**Comment by alandavis** [comment]
RE: markdown. This is not something we wish to encourage, but will be referenced in the docs in acs-packaging.
RE: spring-boot-starter-parent 2.3.6 - Sorry too late. Used the latest version specified in mvnrepository.com. There is always another release. Do them frequently.

**Comment by montgolfiere** [comment]
@alandavis Yes, thanks.

**Pull request merged**